cdp-agentkit-core/typescript/src/actions/cdp/cross_chain_swap.ts

```diff
@@ -0,0 +1,120 @@
+import { CdpAction } from "./cdp_action";
+import { Wallet } from "@coinbase/coinbase-sdk";
+import {
+  createConfig,
+  EVM,
+  getQuote,
+  QuoteRequest,
+  convertQuoteToRoute,
+  executeRoute,
+} from "@lifi/sdk";
+import { base } from "viem/chains";
+import { privateKeyToAccount } from "viem/accounts";
+import { http, createWalletClient, publicActions } from "viem";
+import { z } from "zod";
+
+const CROSS_CHAIN_SWAP_PROMPT = `
+This tool enables cross-chain token swaps using the LiFi protocol. It allows you to find a route to swap tokens from one chain to another. You'll need to specify:
+- The source chain ID
+- The destination chain ID
+- The token on the source chain (address)
+- The token on the destination chain (address)
+- The amount to transfer (in the smallest unit of the token)
+- The address from which the tokens are being transferred
+`;
+
+/**
+ * Input schema for cross-chain swap action.
+ */
+
+export const CrossChainSwapInput = z.object({
+  fromChain: z.number().describe("Source chain ID. e.g. 8453 for Base mainnet"),
+  toChain: z.number().describe("Destination chain ID. e.g. 10 for Optimism mainnet"),
+  fromToken: z
+    .string()
+    .describe(
+      "The contract address of the token on the source chain. Ensure this address corresponds to the specified fromChain.",
+    ),
+  toToken: z
+    .string()
+    .describe(
+      "The contract address of the token on the destination chain. Ensure this address corresponds to the specified toChain.",
+    ),
+  fromAmount: z
+    .string()
+    .describe(
+      "The amount to be transferred from the source chain, specified in the smallest unit of the token (e.g., wei for ETH).",
+    ),
+  fromAddress: z.string().describe("The address from which the tokens are being transferred."),
+});
+
+/**
+ * Executes a cross-chain token swap using the LiFi protocol
+ *
+ * @param wallet - The wallet to execute the swap from
+ * @param args - The swap parameters including chains, tokens, and amounts
+ * @returns A string describing the transaction result
+ */
+export async function crossChainSwap(
+  wallet: Wallet,
+  args: z.infer<typeof CrossChainSwapInput>,
+): Promise<string> {
+  try {
+    const walletAddress = await wallet.getDefaultAddress();
+    const privateKey = walletAddress.export();
+
+    console.log("Wallet address:", walletAddress.getId());
+    console.log("Wallet address type:", typeof walletAddress);
+
+    const client = createWalletClient({
+      account: privateKeyToAccount(privateKey as `0x${string}`),
+      chain: base,
+      transport: http("https://base-mainnet.g.alchemy.com/v2/2yW47qv-8qJlPfJEg0lzt3VPSSR51H2P"),
+    }).extend(publicActions);
```

Comment:
Thanks for the contribution @jlin27 !

We are working on extending AgentKit's wallet support to a more generic interface that will simplify this flow. We anticipate releasing the wallet provider interface in next week's release. I will circle back to this PR once the wallet provider interface is in master.

cc: @murrlincoln

Reply:
Great to hear about the simplified wallet support coming soon!

Will check in and update once it's out. Cheers.